test/FunctionalTests/Microsoft.OData.Core.Tests/NavigationPropertyOnComplexTests.cs

```diff
@@ -344,6 +344,97 @@ public void WriteNavigationPropertyOnComplex()
             Assert.Equal(actual, expected);
         }
 
+        [Fact]
+        public void WriteNavigationPropertyOnComplexFullMetaData()
+        {
+            var uriParser = new ODataUriParser(Model, ServiceRoot, new Uri("http://host/People('abc')?$expand=Address/City($select=ZipCode)"), null);
+            var odataUri = uriParser.ParseUri();
+
+            ODataResource res = new ODataResource() { Properties = new[] { new ODataProperty { Name = "UserName", Value = "abc" } } };
+            ODataNestedResourceInfo nestedComplexInfo = new ODataNestedResourceInfo() { Name = "Address" };
+            ODataResource nestedComplex = new ODataResource() { Properties = new[] { new ODataProperty { Name = "Road", Value = "Zixing" } } };
+            ODataNestedResourceInfo nestedResInfo = new ODataNestedResourceInfo() { Name = "City", IsCollection = false };
+            ODataResource nestednav = new ODataResource() { Properties = new[] { new ODataProperty { Name = "ZipCode", Value = 111 } } };
+
+            string actual = WriteJsonLightEntry(Model, EntitySet, EntityType, odataUri, (writer) =>
+            {
+                writer.WriteStart(res);
+                writer.WriteStart(nestedComplexInfo);
+                writer.WriteStart(nestedComplex);
+                writer.WriteStart(nestedResInfo);
+                writer.WriteStart(nestednav);
+                writer.WriteEnd();
+                writer.WriteEnd();
+                writer.WriteEnd();
+                writer.WriteEnd();
+                writer.WriteEnd();
+            },isFullMetadata:true);
+
+            string expected = "{\"@odata.context\":\"http://host/$metadata#People(Address/City(ZipCode))/$entity\",\"@odata.id\":\"People('abc')\",\"@odata.editLink\":\"People('abc')\",\"UserName\":\"abc\",\"Address\":{\"Road\":\"Zixing\",\"City@odata.associationLink\":\"http://host/People('abc')/Address/City/$ref\",\"City@odata.navigationLink\":\"http://host/People('abc')/Address/City\",\"City\":{\"@odata.id\":\"City(111)\",\"@odata.editLink\":\"City(111)\",\"ZipCode\":111}}}";

```

Comment:
>"{\"@odata.context\":\"http://host/$metadata#People(Address/City(ZipCode))/$entity\",\"@odata.id\":\"People('abc')\",\"@odata.editLink\":\"People('abc')\",\"UserName\":\"abc\",\"Address\":{\"Road\":\"Zixing\",\"City@odata.associationLink\":\"http://host/People('abc')/Address/City/$ref\",\"City@odata.navigationLink\":\"http://host/People('abc')/Address/City\",\"City\":{\"@odata.id\":\"City(111)\",\"@odata.editLink\":\"City(111)\",\"ZipCode\":111}}}"; [](start = 30, length = 453)

can you change the expected string indent?

Reply:
#Resolved